icon_validator/rules/plugin_validators/help_input_output_validator.py

```diff
@@ -4,6 +4,91 @@
 from icon_validator.rules.validator import KomandPluginValidator
 from icon_validator.exceptions import ValidationException
 
+from datetime import datetime
+
+
+def detect_valid_datetime(list_entry: str) -> bool:
+    """
+    This function determines if a valid datetime following
+    iso format is detected in a string
+
+    :param list_entry: The entry of the list to be checked for datetime validity
+    :type list_entry: str
+
+    :rtype: bool
+    """
+    try:
+        datetime.fromisoformat(list_entry)
+        return True
+    except ValueError:
+        return False
+
+
+def convert_to_valid_datetime(date: str) -> str:
+    """
+    This function takes a str containing a date in isoformat and converts
+    it to the format that matches the format found within the raw help md
+
+    :param date: datetime string
+    :type date: str
+
+    :return final_result: datetime string containing 'T'
+    :rtype final_result: str
+    """
+    # Convert the string to a datetime object so we can work with it
+    datetime_object = datetime.fromisoformat(date)
+
+    # Create the datetime containing the T using strftime
+    new_datetime = datetime_object.strftime("%Y-%m-%dT%H:%M:%S")
+
+    # Extract the timezone separately
+    timezone = datetime_object.strftime("%z")
+
+    # If timezone was not found in the string, then just produce the new amended datetime
+    if timezone == '':
+
+        final_result = new_datetime
+
+    else:
+        # Add the ':' in manually because strftime does not contain an option to convert it like this
+        new_timezone = timezone[:3] + ":" + timezone[3:]
```

Comment:
Is it always 3 characters? Or the length should be measured? 😃

Reply:
Good question, I can look into this and see. I think timezone is always in the format of `+DD:DD`